Assets/_Application/Layers/LayerTypes/Polygons/ObjectScatterLayerGameObject.cs

```diff
@@ -362,6 +355,138 @@ public void AddToProperties(RectTransform properties)
+            }
+        }
+
+        private void OnPolygonParentInitialized()
+        {
+            if(polygonLayer.PolygonVisualisation.Bounds == null)
+            {
+                polygonLayer.PolygonVisualisation.OnPolygonVisualisationUpdated.AddListener(OnPolygonVisualisationUpdated);
+            }
+            else
+            {
+                OnPolygonVisualisationUpdated();
+            }            
+            polygonLayer?.OnReferenceChanged.RemoveListener(OnPolygonParentInitialized);
+        }
+
+        private void OnPolygonVisualisationUpdated()
+        {          
+            BoundingBox polygonBoundingBox = polygonLayer.PolygonVisualisation.Bounds;
+            polygonBoundingBox.Convert(CoordinateSystem.RD);
+
+            //check or wait if the maaiveld reference data is loaded
+            RefreshAreaReferenceData(() =>
+            {
+                BinaryMeshLayer bml = ((CartesianTileLayerGameObject)areaReferenceData.Reference).Layer as BinaryMeshLayer;
+                Dictionary<Vector2Int, Tile> activeTiles = bml.tiles;
+                List<BoundingBox> tileBoundingBoxes = new List<BoundingBox>();
+                foreach (Tile tile in activeTiles.Values)
+                    tileBoundingBoxes.Add(GetBoundingBoxForTile(tile));
+
+                if (tileBoundingBoxes.Count > 0 && IsAreaLoadedForBoundingBox(tileBoundingBoxes, polygonBoundingBox, activeTiles.Values.First().layer.tileSize))
+                {
+                    Initialize(LayerData, settings.OriginalPrefabId);
+                }
+                else
+                {
+                    //TODO if a tile is missing for some reason maybe we should do nothing here?
+                    //not all tiles are loaded yet to encapsulate the polygon
+                    bml.OnTileObjectCreated.AddListener(tile =>
+                    {                        
+                        tileBoundingBoxes.Add(GetBoundingBoxForTile(tile));
+                        if (IsAreaLoadedForBoundingBox(tileBoundingBoxes, polygonBoundingBox, tile.layer.tileSize))
+                        {
+                            Initialize(LayerData, settings.OriginalPrefabId);
+                        }
+                    });
+                }
+            });
+
+            polygonLayer.PolygonVisualisation.OnPolygonVisualisationUpdated.RemoveListener(OnPolygonVisualisationUpdated);
+        }
+
+        private void RefreshAreaReferenceData(Action onReferenceDataInitialized)
+        {
+            areaReferenceData = null;
+            List<LayerData> mainLayers = ProjectData.Current.RootLayer.ChildrenLayers;
+            foreach (LayerData l in mainLayers)
+            {
+                if (l is ReferencedLayerData refData)
+                {
+                    if (refData.Reference.Name.Contains(LayerAreaIdentifier))
+                        areaReferenceData = refData;
+                }
+            }
+            if (areaReferenceData != null)
```

Comment:
Zou dit niet in de if van de loop hierboven moeten zitten (en dus een losse methode)? Want als bovenstaande lookup geen resultaten geeft dan is dit altijd null. Tenzij bovenstaande met een Linq query opgelost kan worden

Reply:
Ik zou deze null check verplaatsen naar de Action die meegegeven wordt. Scheelt complexiteit hier en dan heb je een duidelijke guard in die methode - en zelfs de flexibiliteit om iets anders te doen als het toch null is